crates/node/src/network_server/handler/cluster_ctrl.rs

```diff
@@ -45,14 +62,86 @@ impl ClusterCtrlSvc for ClusterCtrlSvcHandler {
             .controller_handle
             .get_cluster_state()
             .await
-            .map_err(|_| tonic::Status::aborted("Node is shutting down"))?;
+            .map_err(|_| Status::aborted("Node is shutting down"))?;
 
         let resp = ClusterStateResponse {
             cluster_state: Some((*cluster_state).clone().into()),
         };
         Ok(Response::new(resp))
     }
 
+    async fn list_logs(
+        &self,
+        _request: Request<ListLogsRequest>,
+    ) -> Result<Response<ListLogsResponse>, Status> {
+        Ok(Response::new(ListLogsResponse {
+            data: serialize_value(self.logs().await?),
+        }))
+    }
+
+    async fn describe_log(
+        &self,
+        request: Request<DescribeLogRequest>,
+    ) -> Result<Response<DescribeLogResponse>, Status> {
+        let request = request.into_inner();
+
+        let log_id = LogId::new(request.log_id);
+        let chain = self
+            .logs()
+            .await?
+            .chain(&log_id)
+            .ok_or(Status::not_found(format!(
+                "Log id {} not found",
+                request.log_id
+            )))?
+            .clone();
+
+        let tail_state = self
+            .bifrost_handle
+            .find_tail(log_id, FindTailAttributes::default())
+            .await
+            .map_err(|err| Status::internal(format!("Failed to find tail: {:?}", err)))?;
+
+        info!("{:?} tail: {:?}", log_id, tail_state);
```

Comment:
pro-tip:
```suggestion
        info!(?tail, ?log_id, "Retrieved tail information");
```

Reply:
Nice! I'd seen this trick before but it's relatively new to me :-)